geth/rpc/call_raw_test.go

```diff
@@ -67,26 +68,57 @@ func TestMethodAndParamsFromBody(t *testing.T) {
 			},
 			"subtract",
 			json.RawMessage(`42`),
+			false,
 		},
 		{
 			"params_empty_array",
 			`{"jsonrpc": "2.0", "method": "test", "params": []}`,
 			[]interface{}{},
 			"test",
 			json.RawMessage(nil),
+			false,
 		},
 		{
 			"params_none",
 			`{"jsonrpc": "2.0", "method": "test"}`,
 			[]interface{}{},
 			"test",
 			json.RawMessage(nil),
+			false,
+		},
+		{
+			"getFilterMessage",
+			`{"jsonrpc":"2.0","id":44,"method":"shh_getFilterMessages","params":["3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e"]}`,
+			[]interface{}{string("3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e")},
+			"shh_getFilterMessages",
+			json.RawMessage(`44`),
+			false,
+		},
+		{
+			"getFilterMessage_array",
+			`[{"jsonrpc":"2.0","id":44,"method":"shh_getFilterMessages","params":["3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e"]}]`,
+			[]interface{}{string("3de6a8867aeb75be74d68478b853b4b0e063704d30f8231c45d0fcbd97af207e")},
+			"shh_getFilterMessages",
+			json.RawMessage(`44`),
+			false,
+		},
+		{
+			"empty_array",
+			`[]`,
+			[]interface{}{},
+			"",
+			json.RawMessage(nil),
```

Comment:
No need to cast `nil` to `json.RawMessage` I guess.

Reply:
@adambabik until we know the nature of this web3.js behaviour, let's assume we just ok to take first element.

Reply:
Somehow `require.Equal` checks for being nil and have the same concrete type. I got error when wrote just `nil`.

Reply:
Ha, interesting. From `reflect.DeepEqual()`:

> Note that a non-nil empty slice and a nil slice (for example, []byte{} and []byte(nil)) are not deeply equal.

`id` is actually an empty slice because it's not a pointer in the struct declaration.

Reply:
You can get rid of `json.RawMessage(nil)` and specify just `nil` if you use `require.EqualValues`.